File: app/src/main/java/com/github/kirillf/materialdemo/SplashActivity.java
package com.github.kirillf.materialdemo;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;

/**
 * Created by k.filimonov on 11/04/2017.
 */

public class SplashActivity extends Activity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.splash_activity);
        new Thread(new Runnable() {
            @Override
            public void run() {
                try {
                    Thread.sleep(2000);
                    onTaskCompleted();
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
            }
        }).start();
    }

    void onTaskCompleted() {
        runOnUiThread(new Runnable() {
            @Override
            public void run() {
                Intent intent = new Intent(SplashActivity.this, MainActivity.class);
                startActivity(intent);
                finish();
            }
        });
    }

}

File: app/src/main/java/com/github/kirillf/materialdemo/SplashActivity.kt
package com.github.kirillf.materialdemo

import android.app.Activity
import android.content.Intent
import android.os.Bundle

/**
 * Created by k.filimonov on 11/04/2017.
 */

class SplashActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.splash_activity)
        Thread(Runnable {
            try {
                Thread.sleep(2000)
                onTaskCompleted()
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }
        }).start()
    }

    private fun onTaskCompleted() {
        runOnUiThread {
            val intent = Intent(this@SplashActivity, MainActivity::class.java)
            startActivity(intent)
            finish()
        }
    }

}
